synapse/lib/agenda.py

```diff
@@ -490,6 +494,17 @@ async def start(self):
         self._schedtask = self.schedCoro(self._scheduleLoop())
         self.enabled = True
 
+    async def stop(self):
+        "Cancel the scheduler loop, and set self.enabled to False."
+        if not self.enabled:
+            return
+        self._schedtask.cancel()
+        await asyncio.sleep(0)
```

Comment:
why do we sleep here?

Reply:
technically, cancel'd tasks haven't been cancelled until they're is a scheduling opportunity in the ioloop to inject the asyncio.Cancelled exeception; but its a bit vestigial here.

Reply:
and i already removed this :D